salt/renderers/gpg.py

```diff
@@ -431,6 +453,12 @@ def render(gpg_data, saltenv="base", sls="", argline="", **kwargs):
         raise SaltRenderError("GPG unavailable")
     log.debug("Reading GPG keys from: %s", _get_key_dir())
 
+    if not __opts__["gpg_decrypt_must_succeed"]:
```

Comment:
Sane defaults 🙏
Nice

Reply:
Maybe
```
if not __opts__.get("gpg_decrypt_must_succeed"):

Reply:
That's the error I'm seeing in tests... key does not exist...

Reply:
right, but I would rather the config be expected to be there.  Since it's defined in `DEFAULT_MASTER_OPTS`, something in the test suite might be overwriting or modifying `__opts__`, in which case the `KeyError` is a sign of a deeper problem.

Reply:
@twangboy I think I misread earlier that the errors were coming from unit tests, not the functional tests.  All other lookups of `__opts__` in `salt/renderers/gpg.py` go through `.get()`, which in my opinion is more dubious than a clean fail if a necessary option is not available, thus the [`configure_loader_modules`](https://docs.saltproject.io/en/latest/topics/development/tests/unit.html#mocking-loader-modules) [comment](https://github.com/saltstack/salt/pull/61418/#issuecomment-1015874055) from @MKLeb .